tests/tests_suite_test.go

```diff
@@ -0,0 +1,165 @@
+		log.Println(err)
+	}
+	controllerCR = schema.GroupVersionResource{
+		Group:    "migration.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "migrationcontrollers",
+	}
+
+	controller, err := dynamicClient.Resource(controllerCR).Namespace(MigrationNamespace).Get(context.TODO(), MigrationController, metav1.GetOptions{})
+	if err != nil {
+		log.Println(err)
+	}
+
+	shouldCreate := true
+	if controller != nil {
+		shouldCreate = false
+	}
+	if shouldCreate {
+		controller, err = dynamicClient.Resource(controllerCR).Namespace(MigrationNamespace).Create(ctx, NewMigrationController(), metav1.CreateOptions{})
+		if err != nil {
+			log.Println(err)
+		}
+	}
+	Eventually(func() string {
+		controller, err := dynamicClient.Resource(controllerCR).Namespace(MigrationNamespace).Get(context.TODO(), MigrationController, metav1.GetOptions{})
+		if err != nil {
+			log.Println(err)
+		}
+		phase, found, err := unstructured.NestedString(controller.Object, "status", "phase")
+		if err != nil || !found {
+			log.Println("phase not found or error in status", err)
+		}
+		return phase
+	}, time.Minute*5, time.Second).Should(Equal("Reconciled"))
+
+	migCluster, secret := NewMigCluster(GetMigSaToken(sourceClient))
+	Expect(hostClient.Create(ctx, secret)).Should(Succeed())
+	Expect(hostClient.Create(ctx, migCluster)).Should(Succeed())
+	Eventually(func() bool {
+		hostClient.Get(ctx, client.ObjectKey{Name: E2ETestObjectName, Namespace: MigrationNamespace}, migCluster)
+		return migCluster.Status.IsReady()
+	}, time.Minute*5, time.Second).Should(Equal(true))
+
+	migStorage, secret := NewMigStorage()
+	Expect(hostClient.Create(ctx, secret)).Should(Succeed())
+	Expect(hostClient.Create(ctx, migStorage)).Should(Succeed())
+	Eventually(func() bool {
+		hostClient.Get(ctx, client.ObjectKey{Name: E2ETestObjectName, Namespace: MigrationNamespace}, migStorage)
+		return migStorage.Status.IsReady()
+	}, time.Minute*5, time.Second).Should(Equal(true))
+
+}, 60)
+
+var _ = AfterSuite(func() {
+
+	ctx := context.TODO()
+	err = hostClient.Delete(ctx, &migapi.MigStorage{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      E2ETestObjectName,
+			Namespace: MigrationNamespace,
+		},
+	})
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = hostClient.Delete(ctx, &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TestStorageSecret,
+			Namespace: ConfigNamespace,
+		},
+	})
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = hostClient.Delete(ctx, &migapi.MigCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      E2ETestObjectName,
+			Namespace: MigrationNamespace,
+		},
+	})
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = hostClient.Delete(ctx, &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TestClusterSecret,
+			Namespace: ConfigNamespace,
+		},
+	})
+	if err != nil {
+		log.Println(err)
+	}
+
+	if shouldCreate {
+		err = dynamicClient.Resource(controllerCR).Namespace(MigrationNamespace).Delete(ctx, MigrationController, metav1.DeleteOptions{})
```

Comment:
question: does deleting the namespace return immediately? 

I know that deletions of namespace sometimes take a lot of time because it triggers deletion of all the objects in the namespace. It might be good to explore the deletion options like Foreground and Background etc to see what is behaviour we want.

Reply:
@alaypatel07 Yes! It does not wait while the ns is being deleted. Move ahead with the rest of the clean-up.

Reply:
if we are planning on creating the same namespace, we will have to wait for deletion for the next test. If we are not then this is fine with me.